core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -318,19 +291,26 @@ private[spark] class MemoryStore(blockManager: BlockManager, memoryManager: Memo
     blockId.asRDDId.map(_.rddId)
   }
 
+  private def acquireStorageMemory(blockId: BlockId, size: Long): Boolean = {
```

Comment:
IIUC this is only called in 1 place. Why bother abstracting it? It's just kind of weird to see that it's releasing pending unroll memory outside the context of putting a block. I actually think this fits better where it used to be.